From pingc@wacom.com Wed Oct  4 09:56:56 2006
From: Eric Sesterhenn <snakebyte@gmx.de>
Subject: USB: fix use after free in wacom_sys.c
Date: Wed, 4 Oct 2006 09:56:44 -0700
Message-ID: <6753EB6004AFF34FAA275742C104F952017568@wacom-nt10.wacom.com>
To: "Greg Kroah-Hartman" <gregkh@suse.de>
Cc: "Eric Sesterhenn" <snakebyte@gmx.de>

From: Eric Sesterhenn <snakebyte@gmx.de>

the following commit added a use after free
http://www.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=3D3bea733ab21247290bd552dd6a2cd3049af9adef
Found by coverity (cid #1441)

Signed-off-by: Eric Sesterhenn <snakebyte@gmx.de>
Signed-off-by: "Ping Cheng" <pingc@wacom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/wacom_sys.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/input/wacom_sys.c
+++ gregkh-2.6/drivers/usb/input/wacom_sys.c
@@ -289,8 +289,8 @@ static void wacom_disconnect(struct usb_
 		input_unregister_device(wacom->dev);
 		usb_free_urb(wacom->irq);
 		usb_buffer_free(interface_to_usbdev(intf), 10, wacom->wacom_wac->data, wacom->data_dma);
-		kfree(wacom);
 		kfree(wacom->wacom_wac);
+		kfree(wacom);
 	}
 }
 
